### PR TITLE
Override INVALID_PARAMETER_VALUE on fetching non-existent job/cluster

### PR DESCRIPTION
**Pull request by mgyucht**
## Changes
Most services use `RESOURCE_DOES_NOT_EXIST` error code with 404 status code to indicate that a resource doesn't exist. However, for legacy reasons, Jobs and Clusters services use `INVALID_PARAMETER_VALUE` error code with 400 status code instead. This makes tools like Terraform and UCX difficult to maintain, as these services need different error handling logic. However, we can't change these behaviors as customers already depend on the raw HTTP response status & contents.

This PR corrects these errors in the SDK itself. SDK users can then do
```go
_, err := w.Jobs.Get(ctx, jobs.GetJobRequest{JobId: 12345})
if errors.Is(err, apierr.ErrResourceDoesNotExist) { ... }
```
just as you would expect from other resources.

Updated the README with more information about this as well.

## Tests
Added unit tests for error overrides.
Added 2 integration tests for Clusters and Jobs.

- [x] `make test` passing
- [x] `make fmt` applied
- [x] relevant integration tests applied

### CODE REVIEW

File: apierr/error_override.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	if e.errorCodeMatcher != nil && !e.errorCodeMatcher.MatchString(err.ErrorCode) {
+		return false
+	}
+	if e.messageMatcher != nil && !e.messageMatcher.MatchString(err.Message) {
+		return false
+	}
+	return true
+}
+
+var invalidParameterValue = "INVALID_PARAMETER_VALUE"
```
Comment (tanmay-db) [inline apierr/error_override.go:54]: If we aren't changing this, then it would be good to have this as a constant

File: apierr/error_override.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+var allOverrides = []errorOverride{
```
Comment (tanmay-db) [inline apierr/error_override.go:56]: nit: Might be good to encapsulate this in a method so it's not accidentally modified.

File: apierr/error_override.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	{
+		debugName:        "Clusters InvalidParameterValue => ResourceDoesNotExist",
+		pathRegex:        regexp.MustCompile(`^/api/2\.0/clusters/get`),
+		verb:             "GET",
+		messageMatcher:   regexp.MustCompile("Cluster .* does not exist"),
+		errorCodeMatcher: regexp.MustCompile(invalidParameterValue),
```
Comment (tanmay-db) [inline apierr/error_override.go:62]: Should we have a `statusCodeMatcher`?

**Review by tanmay-db — APPROVED** [review]
LGTM

File: apierr/error_override.go (hunk 3 above)
Comment (mgyucht) [reply apierr/error_override.go:62]: Good idea, will add.

**Review by tanmay-db — APPROVED** [review]

**Pull request merged**